crates/spk-solve/src/error.rs

```diff
@@ -65,6 +66,10 @@ pub enum Error {
     SolverLogFileIOError(#[source] std::io::Error, PathBuf),
     #[error("Error: Flushing solver log file: {0}")]
     SolverLogFileFlushError(#[source] std::io::Error),
+    #[error(
+        "Cannot build package ({0}) from source during a solve when it has a dependency on itself"
```

Comment:
```suggestion
        "Cannot build package ({0}) from source during a solve because it has a dependency on itself"
```

just want to make sure that this is not always an error - eg if an older version did exist you would be able to form the build environment and proceed with this solve, right?

Reply:
I've updated this.